03week/towersOfHanoi.js

```diff
@@ -19,27 +35,99 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+const movePiece = (startStack,endStack) => {
+  // made  a copy of stacks so i wouldnt mutate the original stacks
+  const stacksCopy = {        
+    a:[],
+    b:[],
+    c:[]
+  
+  }
+  // getting keys from stacks, and copying the value of each key into the copy
+  //using bracket notation to fetch key in string format
+  Object.keys(stacks).forEach( key => {
+    stacksCopy[key] = stacks[key]
+  })
+  // the array with the starting numbers
+  const startingStack = stacksCopy[startStack]
+  // where the block ends up
+  const endingStack = stacksCopy[endStack]
+  // get the value of the last block , so i can move it to the ending stack
+  const currentBlock = startingStack.pop()
+  // pushing it to the ending block
+  endingStack.push(currentBlock);
+  // setting it to the variable the game cares about
+  stacks = stacksCopy
+  
 }
+const  isValid =(startStack)=>{
+  // if input is a,b,or c
+
+  const emptyStartStack = stacks[startStack][stacks[startStack].length === 0];
+  //is the start stack empty? (if it is , === invalid)
+  if (emptyStartStack) {
+    return false
+  }else {
+    return true
+  }
+  // if(stacks[startStack].length == 0){
```

Comment:
You can move a piece to a stack that already has discs, how would you check for this additional rule?